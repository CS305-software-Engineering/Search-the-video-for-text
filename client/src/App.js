import React from "react"
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import SignInSide from "./LoginSignup/login.js"
import HomePage from "./HomePage/homepage.js";
import VideoHistory from "./VideoHistory/videohistory.js";

const App = () => {
  return (
    <Router >
      <Route exact path="/" component={SignInSide} />
      <Route exact path="/home" component={HomePage} />
      <Route exact path="/vhistory" component={VideoHistory} />
    </Router>
  )

}

export default App;
