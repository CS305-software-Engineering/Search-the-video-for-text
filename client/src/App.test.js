import React from 'react';
import { shallow, configure } from 'enzyme';
import App from './App.js';
import {BrowserRouter, Route} from "react-router-dom";

describe('Should render App ', () => {

  it('should render correctly ', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
    component.unmount(); 
  });

  it('should render BrowserRouter correctly', () => {
    const component = shallow(<App />);
    expect(component.find(BrowserRouter)).toHaveLength(1)
    component.unmount()
  });

  it('should render all 3 Route ', () => {
    const component = shallow(<App />);
    expect(component.find(Route)).toHaveLength(3)
    component.unmount()
  });

  it('should have correct route props path', () => {
    const component = shallow(<App />);
    expect(component.find(Route).getElements()[0].props.path).toEqual("/")
    expect(component.find(Route).getElements()[1].props.path).toEqual("/home")
    expect(component.find(Route).getElements()[2].props.path).toEqual("/vhistory")
    component.unmount()
  });

  it('should have correct route props exact property ', () => {
    const component = shallow(<App />);
    expect(component.find(Route).getElements()[0].props.exact).toBeTruthy()
    expect(component.find(Route).getElements()[1].props.exact).toBeTruthy()
    expect(component.find(Route).getElements()[2].props.exact).toBeTruthy()
    component.unmount()
  });

});
