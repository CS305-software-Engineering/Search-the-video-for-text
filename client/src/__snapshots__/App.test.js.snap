// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App  should render correctly  1`] = `
<BrowserRouter>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/home"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/vhistory"
  />
</BrowserRouter>
`;
